tests/unit/test_dimension_utilities.py

```diff
@@ -304,6 +320,165 @@ def test_prefetch_dimension_variables(self, mock_get_opendap_nc4):
+        """ Ensure that the expected bounds array is created given
+            the input dimension variable values.
+
+        """
+        prefetch_dataset = Dataset('tests/data/ATL16_prefetch.nc4', 'r')
+        dimension_path = '/npolar_grid_lat'
+
+        expected_bounds_array = self.bounds_array
+
+        assert_array_equal(get_bounds_array(prefetch_dataset,
+                                            dimension_path),
+                           expected_bounds_array)
+
+    def test_write_bounds(self):
+        """ Ensure that bounds data array is written to the dimension
+            dataset, both when the dimension variable is in the root group
+            and in a nested group.
+
+        """
+        varinfo_prefetch = VarInfoFromDmr('tests/data/ATL16_prefetch_group.dmr')
+        prefetch_dataset = Dataset('tests/data/ATL16_prefetch_group.nc4', 'r+')
+
+        # Expected variable contents in file.
+        expected_bounds_data = self.bounds_array
+
+        with self.subTest('Dimension variable is in the root group'):
+            root_variable_full_path = '/npolar_grid_lat'
+            root_varinfo_variable = varinfo_prefetch.get_variable(
+                root_variable_full_path)
+            root_variable_name = 'npolar_grid_lat'
+            root_bounds_name = root_variable_name + '_bnds'
+
+            write_bounds(prefetch_dataset, root_varinfo_variable)
+
+            resulting_bounds_root_data = prefetch_dataset.variables[
+                root_bounds_name][:]
+            assert_array_equal(resulting_bounds_root_data,
+                               expected_bounds_data)
+            # Check that bounds variable exists in the root group.
+            self.assertTrue(prefetch_dataset.variables[root_variable_name])
+            # Check that bounds variable has expected attributes.
+            # (This is only the 'bounds' attribute, since other attributes
+            # are not required.)
+            root_bounds_varinfo_variable = varinfo_prefetch.get_variable(
+                root_variable_full_path)
+            self.assertEqual(root_bounds_varinfo_variable.attributes['bounds'],
+                             root_bounds_name)
+            # Check that dimension variable has 'bounds' attribute.
+            self.assertEqual(root_varinfo_variable.attributes['bounds'],
+                             root_bounds_name)
+            # Check that VariableFromDmr has 'bounds' reference in
+            # the references dictionary.
+            self.assertEqual(root_varinfo_variable.references['bounds'],
+                             {root_bounds_name, })
+
+        with self.subTest('Dimension variable is in a nested group'):
+            nested_variable_full_path = '/group1/group2/zelda'
+            nested_varinfo_variable = varinfo_prefetch.get_variable(
+                nested_variable_full_path)
+            nested_variable_name = 'zelda'
+            nested_variable_path = '/group1/group2'
```

Comment:
Just spotted this; it's a tiny bit confusing. Would `nested_variable_path` be better as `nested_group_path`?

Reply:
I was thinking of the variable path as being the path to the `zelda` variable, but `nested_group_path` does describe a path made of groups which is the whole point here, so I'll change it.

Reply:
Yeah - not sure if it's just a me thing, but I thought the "path" of a variable was the full thing that gave you the exact location (so including the `zelda` - and being `/group1/group2/zelda`).